geowebcache/mbtiles/src/main/java/org/geowebcache/mbtiles/layer/MBTilesLayer.java

```diff
@@ -0,0 +1,349 @@
+    public String getBlobStoreId() {
+        return null;
+    }
+
+    @Override
+    public boolean isEnabled() {
+        return enabled;
+    }
+
+    @Override
+    public void setEnabled(boolean enabled) {
+        this.enabled = enabled;
+    }
+
+    /** Returns the location of the actual tiles set. */
+    public File getTilesPath() {
+        return tilesPath;
+    }
+
+    public void setTilesPath(File tilesPath) {
+        this.tilesPath = tilesPath;
+    }
+
+    public int getTileSize() {
+        return tileSize;
+    }
+
+    public void setTileSize(int tileSize) {
+        this.tileSize = tileSize;
+    }
+
+    public boolean isVectorTiles() {
+        return vectorTiles;
+    }
+
+    /**
+     * @return {@code true} if success. Note this method's return type should be void. It's not
+     *     checked anywhere
+     * @see org.geowebcache.layer.TileLayer#initialize(org.geowebcache.grid.GridSetBroker)
+     */
+    @Override
+    protected boolean initializeInternal(GridSetBroker gridSetBroker) {
+        if (this.enabled == null) {
+            this.enabled = true;
+        }
+        String specifiedName = getName();
+        if (tilesPath != null) {
+            if (!tilesPath.exists() || !tilesPath.canRead()) {
+                throw new IllegalStateException(
+                        "tilesPath property for this layer "
+                                + (specifiedName != null ? specifiedName : "")
+                                + " is set to '"
+                                + tilesPath
+                                + "' but the file either does not exist or is not readable");
+            }
+        }
+        try {
+            mbTilesFile = new MBTilesFile(tilesPath);
+            cacheInfo = new MBTilesCacheInfo(mbTilesFile);
+            layerBounds = cacheInfo.getBounds();
+            if (StringUtils.isEmpty(specifiedName)) {
+                name = cacheInfo.getMetadataName();
+            }
+        } catch (IOException e) {
+            throw new IllegalStateException("Unable to open the Provided MBTile: " + tilesPath);
+        }
+
+        super.subSets = createGridSubsets(gridSetBroker);
+        super.formats = loadMimeTypes();
+        return true;
+    }
+
+    private List<MimeType> loadMimeTypes() {
+        MBTilesMetadata.t_format metadataFormat = cacheInfo.getFormat();
+        try {
+            switch (metadataFormat) {
+                case PNG:
+                    mimeType = MimeType.createFromFormat("image/png");
+                    break;
+                case JPEG:
+                case JPG:
+                    mimeType = MimeType.createFromFormat("image/jpeg");
+                    break;
+                case PBF:
+                    mimeType = ApplicationMime.mapboxVector;
+                    vectorTiles = true;
+                    break;
+            }
+        } catch (MimeException e) {
+            throw new RuntimeException(e);
+        }
+        if (mimeType != null) {
```

Comment:
Is there any way mimeType can be null  in the above code? Does not seem possible.